src/styles/main.scss

```diff
@@ -1,3 +1,57 @@
-body {
-  margin: 0;
+$cell-size: 100px;
+$gap: 1px;
+$calendar-width: 7 * $cell-size + 6 * $gap;
+$start-margin: 0;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: $calendar-width;
+  padding: 30px 10px;
+  margin: auto;
+
+  &__day {
+    display: flex;
+    width: $cell-size;
+    height: $cell-size;
+    box-sizing: border-box;
+
+    background-color: #eee;
+    border: 1px solid black;
+
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+
+      transform: translateY(-20px);
+    }
+
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+
+        margin: auto;
+
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+  }
+
+  @each $day in mon, tue, wed, thu, fri, sat, sun {
+    &--start-day--#{$day} &__day:first-child {
+      margin-left: $start-margin;
+    }
```

Comment:
create a variable with days of the week and their start margin and use it inside @each

https://sass-lang.com/documentation/at-rules/control/each